### DISCUSSION TITLE
RAG APPLICATION SUGGESTION NEEDED

### DISCUSSION THREAD

**Discussion by kashh56** [flet-dev/flet#4588] [General]
hey ...i am new here i dont have any prior experience to building applications , I have basic knowledge about  FLask, django , html ,css , ai/ml deep learning rag based applications, I want to build a rag based **cross platform application** that uses  custom data for fashion recommendation on the users query.. it uses open source llm and embedding model , its basically RAG based can i build this using flet or any other framework will be better i want to deploy it as well on the playstore and appstore , i only know python ...

**Comment by ap4499** [comment]
Presumably you're targeting **_only_** iOS and Android?

What is the plan for the model, you will be running via an API to a web based platform?

If you are considering running the models locally, on a phone.. I think that is going to be pretty tough given mobile operating systems.. you would likely need to use something like onnx, or TFLite. I think Apple use Core ML. (I think the Pixel uses TFLite for Gemini Nano)

As for Flet as a framework.. I think that depends on the above, and whether it is run locally or via a server API.

**Reply by ap4499** [reply]
After a bit of research, Flutter does have tflite_flutter and core_ml_flutter, to help with the task on each OS respectively.

However, I don't believe either are implemented within Flet.
